WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -196,6 +196,70 @@ def register
     end
   end
 
+  def process_payment
+    competition = competition_from_params
+    if current_user
+      registrations = competition.registrations
+      registration = registrations.find_by_user_id!(current_user.id)
+    end
+    token = params[:stripeToken]
+    Stripe.api_key = ENVied.STRIPE_API_KEY
+
+    begin
+      charge = Stripe::Charge.create({
+        amount: registration.outstanding_entry_fees.cents,
```

Comment:
hmm, in the normal application flow yes. If someone can post to this URL without `current_user` set then no.